topics/ICRC-21/ICRC-21.did

```diff
@@ -0,0 +1,142 @@
+// Canister interface specification for ICRC-21.
+// See https://github.com/dfinity/wg-identity-authentication/blob/main/topics/ICRC-21/icrc_21_consent_msg.md
+
+type icrc21_consent_message_spec = record {
+    // BCP-47 language tag. See https://www.rfc-editor.org/rfc/bcp/bcp47.txt
+    language: text;
+    // Information about the device responsible for presenting the consent message to the user.
+    device_spec: variant {
+        // A generic display able to handle large documents and do line wrapping and pagination / scrolling.
+        // Text must be Markdown formatted, no external resources (e.g. images) are allowed.
```

Comment:
nit: since we already are having variants, maybe it would make sense to explicitly distinguish also between `GenericDisplayText` and `GenericDisplayMarkdown`, so that FE-developers don't have to guess (or rely on some library's guess) whether the message is in Markdown or not?

Reply:
The idea is to restrict the flexibility. I.e. `GenericDisplayText` is _always_ Markdown and `LineDisplayMessage` must be plain text.

That way developers don't have to deal with 4 options.